### DISCUSSION TITLE
opensumi如何存储自定义的配置信息

### DISCUSSION THREAD

**Discussion by hanbingzi** [opensumi/core#2384] [Q&A]
我开发了一些功能，需要存储一些自定义的配置信息，方便下次启动时进行读取，应该如何才能做到那？

**Comment by erha19** [comment]
@hanbingzi 可以使用本地存储服务 `StorageProvider`, 需要你手动创建一下存储路径，如：

https://github.com/opensumi/core/blob/75511a4849d2dde6936616bccd3c8d987326d62f/packages/core-common/src/storage.ts

这里的 `STORAGE_SCHEMA.SCOPE` 最终数据文件会存储在 `~/.sumi/datas/xxx` 下面，而 `STORAGE_SCHEMA.GLOBAL` 会直接存储在 `~/.sumi` 下

用法可以参考：

https://github.com/opensumi/core/blob/4854f4fe3a3117dcbfca0292497c9a69849c8e3c/packages/file-tree-next/src/browser/services/file-tree-model.service.ts#L412

**Comment by hanbingzi** [comment]
谢谢
